packages/flutter/lib/src/foundation/object.dart

```diff
@@ -2,19 +2,26 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// @dart = 2.8
-
 /// Framework code should use this method in favor of calling `toString` on
 /// [Object.runtimeType].
 ///
 /// Calling `toString` on a runtime type is a non-trivial operation that can
 /// negatively impact performance. If asserts are enabled, this method will
 /// return `object.runtimeType.toString()`; otherwise, it will return the
 /// [optimizedValue], which must be a simple constant string.
-String objectRuntimeType(Object object, String optimizedValue) {
+String objectRuntimeType(Object? object, String optimizedValue) {
   assert(() {
     optimizedValue = object.runtimeType.toString();
     return true;
   }());
   return optimizedValue;
 }
+
+/// Allows promotion of a variable to another type.
```

Comment:
This feels like a strange thing to add to flutter's public API. Looks like it is only used in one file. Can we just add it as a private thing there? Or, instead of using promote, can you just do the following in the code:

```
num n = 42;
n = n as int;
n.isEven;
```

Reply:
I don't remind the issue but @Hixie suggested to add this function.

Reply:
Finally I removed this function for now.